src/tools/perf/lib/uct_tests.cc

```diff
@@ -773,7 +773,7 @@ class uct_perf_test_runner {
    TEST_CASE_ALL_OSD(_perf, _case, UCT_PERF_DATA_LAYOUT_BCOPY) \
    TEST_CASE_ALL_OSD(_perf, _case, UCT_PERF_DATA_LAYOUT_ZCOPY)
 
-ucs_status_t uct_perf_test_dispatch(ucx_perf_context_t *perf)
+ucs_status_t __attribute__ ((optimize("O1"))) uct_perf_test_dispatch(ucx_perf_context_t *perf)
```

Comment:
why is attribute needed here?

Reply:
Without optimization the stack frame size > 8192 bytes. I tried using "new" at "src/tools/perf/lib/ucp_tests.cc" to make the runner use the heap instead of the stack and it compiled successfully on the HPC node, but then the CI failed because of the "new" and I couldn't reproduce it.
At the moment I changed it to pragma instead of the attribute.

Reply:
Outdated, done in other PRs